### PR TITLE
trello-bug-297: Autoposters - Strip out bad code in ads

### PR DESCRIPTION
**Pull request by nasimoyz**
https://trello.com/c/xbVErAO7/297-5-autoposters-strip-out-bad-code-in-ads
This PR makes some improvements to sanitization that CS has requested.

Improvements:
* Replaces strong tags that appear at the very start of the document
* Replace strong tags with h3 when preceded by closing ul
* Account for whitespace in PsuedoHeadings

### CODE REVIEW

File: src/Filters/ConvertPsuedoHeadings.php [hunk 1]
```diff
@@ -20,14 +20,26 @@ public function apply(string $html): string
             $html
         );
 
+        $html = preg_replace(
+            '/^(<strong>(.{0,80}?)<\/strong>(?:\s+)?(<br *\/?>)?(?:\s+)?)+/si',
```
Comment (developerdino) [inline src/Filters/ConvertPsuedoHeadings.php]: I see that this is from before but why only 80 characters? Seems like an odd boundary!

Comment (developerdino) [reply src/Filters/ConvertPsuedoHeadings.php]: seems like `(.{0,80}?)` can be replaced by `(.*?)` without any issues as it is non greedy

Comment (nasimoyz) [reply src/Filters/ConvertPsuedoHeadings.php]: I guess we wanted to impose a limit to the heading length on jobs coming in from Autoposter. I agree it looks somewhat arbitrary, but felt it was out of scope of the story to update them

Comment (developerdino) [reply src/Filters/ConvertPsuedoHeadings.php]: Yeah definitely out of scope

**Pull request merged**